stable/prometheus/templates/server-configmap.yaml

```diff
@@ -14,32 +14,5 @@ data:
 {{- range $key, $value := .Values.serverFiles }}
   {{ $key }}: |
 {{ toYaml $value | default "{}" | indent 4 }}
-{{- if eq $key "prometheus.yml" -}}
-{{- if $root.Values.alertmanager.enabled }}
```

Comment:
Removing this section will always enable alertmanager service discovery even if the user use `alermanager.enabled: fals`.

Reply:
It will also disable the ability to customise prometheus.yml

Reply:
@gianrubio , thanks for the review.

This change is indeed a breaking change for users who would expect that their custom `values.yaml` file will work with newer versions of the chart. However, keeping the alertmanager configuration inside the `server-configmap.yaml` is restricting and does not cover all use-cases.

> Removing this section will always enable alertmanager service discovery even if the user use alermanager.enabled: false.

I've moved the section in the `values.yaml`, so not always but it would require a couple of changes if a user wants to disable alermanager - set the property to false and also remove the alertmanager configuration from the `values.yml` file. I figured that this kind of configuration is already precedented by the scrape configs in the config file. 

If the goal is to manage the alertmanager with a single property, I suppose this change should not get merged. Please advice.

Reply:
I don't see how this removes the ability to customize `prometheus.yaml`. It just removes a no longer needed key check in the range. The `prometheus.yaml` key will still be mapped from `.Values.serverFiles`.

Reply:
The custom prometheus.yaml seems to be not customizable /metric I define the consul_sd_configs field. Err:field targets not found in type consul.plain